File: Java Implementation/Edge.java

public class Edge {
    public Node start;
    public Node end;
    public double weight;

    public Edge(Node start, Node end) {
        this.start = start;
        this.end = end;
        this.weight = 1.0;
    }

    public Edge(Node start, Node end, double weight) {
        this.start = start;
        this.end = end;
        this.weight = weight;
    }
    
    public String toString() {
    	return start.toString() + " " + end.toString() + " " + weight;
    }
    
}

File: C++ Implementation/Edge.cpp
#ifndef CLASS_EDGE
#define CLASS_EDGE


#include "Edge.hpp"


Edge::Edge(Node* _start, Node* _end) : start(_start), end(_end), weight(1.0) {}
Edge::Edge(Node* _start, Node* _end, float _weight) : start(_start), end(_end), weight(_weight) {}
Edge::Edge(const Edge &_edge) : start(_edge.start), end(_edge.end), weight(_edge.weight) {}

#endif